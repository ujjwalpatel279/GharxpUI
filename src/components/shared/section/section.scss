// ========================================================================
// COMPONENTS/SHARED/SECTION
// ========================================================================

@import '~sassHelpers';

.c-section {
  background: white;
  border: 1px solid color(grey4);
  border-radius: $border-radius;
  box-shadow: $box-shadow;

  &__header {
    display: flex;
    flex-direction: column;
    height: spacer(25);
    padding-left: spacer(5);
    padding-right: spacer(5);
    padding-top: spacer(5);
    padding-bottom: spacer(5);
    border-bottom: 1px solid color(grey4);

    h3 {
      font-size: font-size(3);
      margin-bottom: spacer(3);
    }

    @include bp(20) {
      align-items: center;
      flex-direction: row;
      height: spacer(16);

      h3 {
        margin: 0;
      }

      div {
        margin-left: auto;
      }
    }
  }

  &__body {
    padding: spacer(5);
  }
}
