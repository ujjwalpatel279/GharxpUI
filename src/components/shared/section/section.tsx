import React, { FunctionComponent, ReactElement } from 'react';

import './section.scss';

interface SectionProps {
  title?: string;
  headerChildren?: ReactElement;
  mb?: 0 | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8;
}

export const Section: FunctionComponent<SectionProps> = ({ title, headerChildren, children, mb }): ReactElement => {
  let sectionClasses = 'c-section';

  if (mb) {
    sectionClasses += ` u-mb-${mb}`;
  }

  return (
    <section className={sectionClasses}>
      <header className="c-section__header">
        <h3>{title}</h3>
        {headerChildren}
      </header>
      <div className="c-section__body">{children}</div>
    </section>
  );
};
