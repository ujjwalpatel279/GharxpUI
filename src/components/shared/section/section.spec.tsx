import React from 'react';
import { render, cleanup } from '@testing-library/react';

import { Section } from './section';

describe('Section Component', () => {
  afterEach(cleanup);

  it('renders the section component', () => {
    const { container } = render(<Section title="Section">Container Component</Section>);
    expect(container.firstChild).toMatchSnapshot();
  });
});
