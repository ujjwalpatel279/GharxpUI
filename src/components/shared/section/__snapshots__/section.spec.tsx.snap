// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section Component renders the section component 1`] = `
<section
  class="c-section"
>
  <header
    class="c-section__header"
  >
    <h3>
      Section
    </h3>
  </header>
  <div
    class="c-section__body"
  >
    Container Component
  </div>
</section>
`;
